evmcore/tx_pool_test.go

```diff
@@ -3134,6 +3135,49 @@ func TestSampleHashesManySenders(t *testing.T) {
 	}
 }
 
+func TestTxPool_ResetDropsTransactionsWithHighGas(t *testing.T) {
```

Comment:
Please also add a check that the filtering is not conducted if no change in the state of the Osaka enabled flag. Thus, most of the time, this cleanup should not be conducted.

Effectively, this cleanup will only be run once, right? The first time there is a re-org after the Osaka hard-fork?